### PR TITLE
docs: --ignore-build-cache -> --no-run-cache

### PR DESCRIPTION
**Pull request by efiop**
Per https://github.com/iterative/dvc/pull/3788

> You may disregard these recommendations if you used the **Edit on GitHub** button from dvc.org to improve a doc in place.

❗ Please read the guidelines in the [Contributing to the Documentation](https://dvc.org/doc/user-guide/contributing/docs) list if you make any substantial changes to the documentation or JS engine.

🐛 Please make sure to mention `Fix #issue` (if applicable) in the description of the PR. This causes GitHub to close it automatically when the PR is merged.

Please chose to [allow us](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to edit your branch when creating the PR.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

File: content/docs/command-reference/repro.md [hunk 1]
```diff
@@ -127,7 +127,7 @@ and only execute the final stage.
 - `-P`, `--all-pipelines` - reproduce all pipelines, for all the stage files
   present in `DVC` repository.
 
-- `--ignore-build-cache` - in cases like `... -> A (changed) -> B -> C` it will
+- `--ignore-run-cache` - in cases like `... -> A (changed) -> B -> C` it will
```
Comment (shcheklein) [inline content/docs/command-reference/repro.md:130]: do we still have the semantics/description considering that we changed the caching mechanism? (stage vs the whole run) ... is there a summary somwhere btw what exactly do we and how?

**Comment by shcheklein** [comment]
update zsh/bash?

update help if needed?

File: content/docs/command-reference/repro.md (hunk 1 above)
Comment (efiop) [reply content/docs/command-reference/repro.md:130]: Yeah, thinking about maybe introducing a separate option for the previous behaviour.

Comment (efiop) [reply content/docs/command-reference/repro.md:130]: Might also use `--no-run-cache`, seems more natural.

**Comment by efiop** [comment]
@shcheklein Updated the original PR and the docs. Now we have `--no-run-cache`(nice symmetry with `--run-cache` for push/pull/etc, though it is undocumented in those right now) + `--force-downstream`)

File: content/docs/command-reference/repro.md [hunk 2]
```diff
@@ -127,7 +127,10 @@ and only execute the final stage.
 - `-P`, `--all-pipelines` - reproduce all pipelines, for all the stage files
   present in `DVC` repository.
 
-- `--ignore-build-cache` - in cases like `... -> A (changed) -> B -> C` it will
+- `--no-run-cache` - Run changed stage even if it has been already ran with the
```
Comment (efiop) [inline content/docs/command-reference/repro.md:130]: Very sloppy explanation copied from the --help :slightly_frowning_face: https://github.com/iterative/dvc/pull/3788#discussion_r424051444

File: content/docs/command-reference/repro.md [hunk 3]
```diff
@@ -127,7 +127,10 @@ and only execute the final stage.
...
+  same command/dependencies/outputs/etc before.
+
+- `--force-downstream` - in cases like `... -> A (changed) -> B -> C` it will
```
Comment (shcheklein) [inline content/docs/command-reference/repro.md:133]: yep, sounds better!

**Comment by shcheklein** [comment]
@efiop I like the names better! Could you summarize please - what do we cache exactly in the run cache now?

File: content/docs/command-reference/repro.md (hunk 2 above)
Comment (shcheklein) [reply content/docs/command-reference/repro.md:130]: Run -> run

good for now, I think ... we'll def need to explain the whole concept of the run cache here on in the UG - it has a lot of moving pieces. It'll be easier to explain this option better after that

please, create a ticket with some summary of the run cache - what commands does it affect, etc?

File: content/docs/command-reference/run.md [hunk 4]
```diff
@@ -161,8 +161,8 @@ data pipeline (e.g. random numbers, time functions, hardware dependency, etc.)
   determined by the logic described in the `-f` option) without asking for
   confirmation.
 
-- `--ignore-build-cache` - forcefully execute the `command` again, even if the
-  same `dvc run` command has already been run in this workspace. Useful if the
+- `--no-run-cache` - forcefully execute the `command` again, even if the same
+  `dvc run` command has already been run in this workspace. Useful if the
```
Comment (shcheklein) [inline content/docs/command-reference/run.md:165]: it is not this workspace anymore, right?

Comment (efiop) [reply content/docs/command-reference/run.md:165]: Yeah, but I have trouble explaining it here better without defining run-cache before that. Looks like I should create `run-cache` entry in our user guide to explain it. But `--run-cache` is hidden in `push/pull/etc` right now, so we can keep `workspace` in this explanation, as without those options you can't really transfer run-cache anywhere.

Comment (shcheklein) [reply content/docs/command-reference/run.md:165]: yeah, make it a checkbox in the ticket to write and update stuff related to the run cache.

File: content/docs/command-reference/repro.md (hunk 2 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/repro.md:130]: Please see https://github.com/iterative/dvc/pull/3788#pullrequestreview-411108828

Comment (efiop) [reply content/docs/command-reference/repro.md:130]: @shcheklein Created https://github.com/iterative/dvc.org/issues/1289
